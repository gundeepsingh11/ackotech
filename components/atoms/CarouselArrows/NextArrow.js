// @flow
/**
 * NextArrow component to represent the right arrow for carousel
 * @module NextArrow
 */
import React, { Fragment } from 'react';
import withStyles from '../../../lib/withStyles';
import styles from './Arrow.style';

const NextArrow = ({ onClick, className }) => (
  <Fragment>
    <button
      onClick={onClick}
      className={`icon-right-arrow-large arrow-small ${className}`}
      type="button"
    />
  </Fragment>
);

export default withStyles(NextArrow, styles);
export { NextArrow as NextArrowVanilla };
