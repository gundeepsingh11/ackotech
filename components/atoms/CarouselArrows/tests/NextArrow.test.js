import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { NextArrowVanilla } from '../NextArrow';

const props = {
  onClick: jest.fn(),
  className: 'arrow',
};
const openNavSpy = jest.fn();

describe('<NextArrow />', () => {
  let Component = '';
  beforeEach(() => {
    Component = shallow(<NextArrowVanilla {...props} />);
  });

  test('should render correctly', () => {
    expect(Component).toMatchSnapshot();
  });

  test('should render correctly', () => {
    Component = shallow(<NextArrowVanilla onClick={openNavSpy} />);
    expect(Component).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    Component = renderer.create(<NextArrowVanilla />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
