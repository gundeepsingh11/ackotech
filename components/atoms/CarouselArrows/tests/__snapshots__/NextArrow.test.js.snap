// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextArrow /> should render correctly 1`] = `
<Fragment>
  <button
    className="icon-right-arrow-large arrow-small arrow"
    onClick={[MockFunction]}
    type="button"
  />
</Fragment>
`;

exports[`<NextArrow /> should render correctly 2`] = `
<Fragment>
  <button
    className="icon-right-arrow-large arrow-small undefined"
    onClick={[MockFunction]}
    type="button"
  />
</Fragment>
`;

exports[`<NextArrow /> should render with styles correctly and match snapshot 1`] = `
<button
  className="icon-right-arrow-large arrow-small undefined"
  type="button"
/>
`;
